packages/editor/src/components/provider/index.js

```diff
@@ -193,17 +193,19 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 					hasLoadedPostObject: _hasLoadedPostObject,
 					editorSettings: getEditorSettings(),
 					isReady: __unstableIsEditorReady(),
-					mode: getRenderingMode(),
+					mode: template ? getRenderingMode() : 'post-only',
 					defaultMode:
-						postTypeObject?.default_rendering_mode ?? 'post-only',
+						template && postTypeObject?.default_rendering_mode
+							? postTypeObject?.default_rendering_mode
+							: 'post-only',
 					selection: getEditorSelection(),
 					postTypeEntities:
 						post.type === 'wp_template'
 							? getEntitiesConfig( 'postType' )
 							: null,
 				};
 			},
-			[ post.type ]
+			[ template, post.type ]
```

Comment:
Can we pass `hasTemplate` or a similar boolean value? There is no need to re-run `mapSelect` after every template change.

It's also odd that this line resolves true when `template` isn't available. Let's check if the template is passed as an empty object when missing.

https://github.com/WordPress/gutenberg/blob/79200bfea601b51b7cfc5414a62e56aca63509be/packages/editor/src/components/provider/index.js#L209